server/src/main/java/org/elasticsearch/index/search/MatchQuery.java

```diff
@@ -360,7 +360,7 @@ protected Query analyzePhrase(String field, TokenStream stream, int slop) throws
                     throw exc;
                 }
             }
-            Query query = super.analyzePhrase(field, stream, slop);
+            Query query = mapper.analyzePhrase(field, stream, slop);
```

Comment:
`analyzePhrase` is called when there are no synonyms in the query so I think we should also override `analyzeMultiPhrase` to handle single term synonyms. Since the fixed size shingle filter doesn't handle multi-terms synonyms there is no need to override `analyzeGraphPhrase` which should continue to use the main field.